lib/datadog/open_feature/binding/configuration.rb

```diff
@@ -0,0 +1,292 @@
+            splits: parse_splits(allocation_data['splits'] || []),
+            do_log: allocation_data.fetch('doLog', true)
+          )
+        end
+
+        private
+
+        def self.parse_rules(rules_data)
+          return nil if rules_data.nil? || rules_data.empty?
+
+          rules_data.map { |rule_data| Rule.from_json(rule_data) }
+        end
+
+        def self.parse_splits(splits_data)
+          splits_data.map { |split_data| Split.from_json(split_data) }
+        end
+
+        def self.parse_timestamp(timestamp_data)
+          return nil if timestamp_data.nil?
+
+          # Handle both Unix timestamps and ISO8601 strings
+          case timestamp_data
+          when Numeric
+            Time.at(timestamp_data)
+          when String
+            Time.parse(timestamp_data)
+          else
+            nil
+          end
+        rescue StandardError
+          nil
+        end
+      end
+
+      # Represents a traffic split within an allocation
+      class Split
+        attr_reader :shards, :variation_key, :extra_logging
+
+        def initialize(shards:, variation_key:, extra_logging: nil)
+          @shards = shards || []
+          @variation_key = variation_key
+          @extra_logging = extra_logging || {}
+        end
+
+        def self.from_json(split_data)
+          new(
+            shards: parse_shards(split_data['shards'] || []),
+            variation_key: split_data['variationKey'],
+            extra_logging: split_data['extraLogging'] || {}
+          )
+        end
+
+        private
+
+        def self.parse_shards(shards_data)
+          shards_data.map { |shard_data| Shard.from_json(shard_data) }
+        end
+      end
+
+      # Represents a shard configuration for traffic splitting
+      class Shard
+        attr_reader :salt, :total_shards, :ranges
+
+        def initialize(salt:, total_shards:, ranges:)
+          @salt = salt
+          @total_shards = total_shards
+          @ranges = ranges || []
+        end
+
+        def self.from_json(shard_data)
+          new(
+            salt: shard_data['salt'],
+            total_shards: shard_data['totalShards'],
+            ranges: parse_ranges(shard_data['ranges'] || [])
+          )
+        end
+
+        private
+
+        def self.parse_ranges(ranges_data)
+          ranges_data.map { |range_data| ShardRange.from_json(range_data) }
+        end
+      end
+
+      # Represents a shard range for traffic allocation
+      class ShardRange
+        attr_reader :start, :end_value
+
+        def initialize(start:, end_value:)
+          @start = start
+          @end_value = end_value
+        end
+
+        def self.from_json(range_data)
+          new(
+            start: range_data['start'],
+            end_value: range_data['end']
+          )
+        end
+
+        # Alias for backward compatibility
+        def end
+          @end_value
+        end
+      end
+
+      # Represents a targeting rule
+      class Rule
```

Comment:
`TargetingRule`?

Reply:
Also matching the field name in the universal flag configuration

https://github.com/DataDog/dd-trace-rb/blob/49dafd36f25491fa1107d659ca2aa4578fd11c49/spec/fixtures/ufc/flags-v1.json#L813